### DISCUSSION TITLE
No connection route from first biome to second biome for seed "b@@`cBj?" (maybe #10047 again)

### DISCUSSION THREAD

**Discussion by kaedepon** [FakeFishGames/Barotrauma#13791] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Maybe it is again of #10047 
Start seed is ```b@@`cBj?```
The first biome has no traversable route connecting it to the second biome for the campaign.

----
![missingpath](https://github.com/FakeFishGames/Barotrauma/assets/47914478/50e4462e-7143-4410-9cbf-a02b63439140)

The save data is here.
[No_connection.zip](https://github.com/FakeFishGames/Barotrauma/files/15043071/No_connection.zip)

### Reproduction steps

1. Input Seed ```b@@`cBj?```
2. Start a campaign.
3. Watch generated map.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.3.0.4

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
Is this real seed generated by game?
How did you find it?

**Reply by kaedepon** [reply]
Yes, this is not a real seed generated by the game.
However, a real seed with the same map could be generated by the game.

I found this when I was building and running a program to examine maps for speedrun by brute force.

**Reply by Regalis11** [reply]
>Is this real seed generated by game?

It shouldn't really matter - a "weird seed" should not lead to weird maps, so this does seem like something we do need to investigate. :)

@kaedepon out of curiosity, do you have some rough idea of how many maps the program needed to generate before you ran into this? E.g. was it something like the tenth map you tested, or did you analyze something like hundreds or thousands of maps? Just trying to get a rough idea how common this might be.

**Reply by kaedepon** [reply]
At this time, I have examined about 130,000 maps and found 5 seeds, including this report.
So we will encounter them in about 0.0038% of the cases.

Additional Information:
I found remaining 4 seeds is
- ```0000PPP@ace]```
- ```00`PPP@Pad3F```
- ```00@@PP@`ad]Q```
- ```00`0PP`Paf8k```

**Comment by SomeRandomNoobKekeke** [comment]
Do you check just connectivity?
Because i've seen some weird seeds in dynamic europa where you start in red zone in the middle of the map, or where end outpost is abandoned
I'm not sure if this could happen in vanilla though

**Reply by kaedepon** [reply]
Yes, for now.
I am looking for a seed that is easy to use with the shortest route from start to finish.
> Because i've seen some weird seeds in dynamic europa where you start in red zone in the middle of the map, or where end outpost is abandoned

I don't know about dynamic europa, but if it is vanilla, I have seen source code that prevents such problems.

**Comment by Regalis11** [comment]
I've created [a ticket about this](https://github.com/FakeFishGames/Barotrauma/issues/13797) in the issue tracker now, we'll look into it!

**Comment by Regalis11** [comment]
Fixed in our private repo, the fix should be coming up in the next unstable build.
